### DISCUSSION TITLE
Timezones are being ignored on iOS

### DISCUSSION THREAD

**Discussion by rachel-carvalho** [ente-io/ente#807] [Enhancements]
**Describe the bug**
When a photo taken on one timezone (ie UTC -3) is uploaded on another timezone (ie UTC -5), its timestamp is the original one but uses the current device's timezone.

So if a photo is taken at 13:00 UTC -3 (16:00 UTC), it's being displayed as 13:00 UTC -5 when viewing on this time zone, instead of 11:00 which is the correct timestamp.

I'm not sure if this is an issue on iOS only or not.

**To Reproduce**
1. Take a photo on a different timezone than your current
2. Upload it
3. See how the timestamp shows your current timezone instead of the originally taken one

**Expected behavior**
The timestamp should remain correct regardless of timezones. Ideally we'd see the original time in the original timezone.

**Screenshots**
I'm adding screenshots below.

**Smartphone (please complete the following information):**
 - Device: iPhone 13
 - OS: iOS 17.2.1
 - App Version 0.8.48

**Additional context**
I noticed this because another user who I share an album with has photos (taken and uploaded on Android) that should be within minutes of mine but are apart by 2 hours, the difference between my current timezone and the one from the photos. Their photos seem to have the correct timestamp. 

✅ Here are screenshots with the details + EXIF of the photo with correct timestamps, taken and uploaded on Android:

<img src="https://github.com/ente-io/photos-app/assets/446046/fafc36ed-5023-4346-8b20-7c8065b4aa8c" alt="android-correct-timestamp-details" width="300px">

<img src="https://github.com/ente-io/photos-app/assets/446046/04e4110e-e21e-46ea-a263-d9573764cadb" alt="android-correct-timestamp-exif" width="300px">

❌ Details + EXIF of the iOS photo with incorrect timestamp:

<img src="https://github.com/ente-io/photos-app/assets/446046/5d3c1387-0475-454d-9649-7aceee59f611" alt="ios-wrong-timestamp-details" width="300px">

<img src="https://github.com/ente-io/photos-app/assets/446046/3eb1221a-2fcb-4664-b267-0320c3d3cbf4" alt="ios-wrong-timestamp-exif1" width="300px">

<img src="https://github.com/ente-io/photos-app/assets/446046/ecf58d21-1742-4549-94aa-a795736393a8" alt="ios-wrong-timestamp-exif2" width="300px">

**Comment by ua741** [comment]
Hello, 
We had released a potential fix for this issue on our mobile app. Can you please confirm if you are still facing this issue? For existing photos, you can delete them from Ente and re-upload if they are still present.

If that's not an option, we will modify our desktop/web app bulk change time API to handle this particular case.

**Comment by gkoz** [comment]
@ua741 

I'd like to second the request that the app show the local time of the photos.

I've just had a similar issue with the desktop app: when uploading photos with no EXIF time offset it appears to assume the system timezone of the app. It's a fine default but
- the time zone is not displayed in the desktop app;
- the "Fix time" tool doesn't allow adjusting it and is not aware of time zones at all.

Given that EXIF doesn't appear to record the time zone, only the offset, it would seem even more important to allow editing the time zone and perhaps prompt the user to do so when the EXIF offset is inconsistent with the assumed time zone.

**Comment by codethief** [comment]
I'm having the same issue on Android and in the desktop app: I took some photos in locations with timezone UTC-5. Some of the photos I uploaded while still there, the rest of the photos I uploaded back home (UTC+1). Both Android and desktop app, however, seem to interpret the photos' local times as times in my current timezone (UTC+1). As a result, in the photo overview those photos I took in the evening are shown as having been taken the next day in the morning etc.

Nevertheless, the Ente app's info button for any given individual photo does show the correct local time, **including the timezone offset** (@gkoz), e.g.

```
ExifVersion: 0232
DateTimeOriginal: 2025:08:24 11:27:40
DateTimeDigitized: 2025:08:24 11:27:40
OffsetTime: -05:00
OffsetTimeOriginal: -05:00
OffsetTimeDigitized: -05:00
```

@rachel-carvalho's screenshot showed similar EXIF data. So could it be that, whether or not the offset is stored in the EXIF data, depends on the camera app in question?

In my case, I use a Google Pixel w/ Google Camera. Even if the offset weren't present in the EXIF data, since my photos' filenames `PXL_YYYYMMDD_HHMMSS….jpg` indicate the time in UTC, if push comes to shove, the timezone offset could also be determined by comparing the EXIF data's local time to the timestamp in the filename.


PS: @rachel-carvalho Would you mind renaming this thread and removing the "on iOS" part since the issues doesn't seem platform-specific?

**Comment by angadsri** [comment]
Did anyone manage to fix this issue, without reuploading the offending photos? @rachel-carvalho @ua741 

I assume the bulk-change feature is still web/desktop-only (which I will try to use when my rather big library loads up in Safari on my iPad) — but I don't know if it can/should be attempted on all of my photos at once. 

Edit: trying the bulk-change on desktop for about 500 photos, hopefully it goes well then I can try with more. I guess you can ignore my update to this thread given this feature exists—save perhaps to dissuade me if using the bulk-time-change on my 80000+ photos at once is a bad idea 😛

**Reply by ua741** [reply]
@angadsri Are you still facing this issue? To best of my understanding, both web and mobile client displays correct time in the file info panel for such photos, at least for newer uploads in last 1 year or so.

**Reply by angadsri** [reply]
Good day Neeraj @ua741 -- yes unfortunately this issue is occurring with photos clicked in June 2025, exported from Google Photos and uploaded well after that in Feb 2026

**Reply by neeraj-pilot** [reply]
@angadsri Can you please write to support@ente.com with more details, including screenshot + link to sample file if possible. We will investigate.

**Reply by angadsri** [reply]
Hi @neeraj-pilot, just sent the email, thanks for your patience!
